### DISCUSSION TITLE
Differences between zoxide and z/z.lua?

### DISCUSSION THREAD

**Discussion by c02y** [ajeetdsouza/zoxide#149] [Q&A] [answered]
Hi, you said zoxide is inspired by z and z.lua.

I've using z.lua for about 2 years, just found zoxide, since I'm very interested in Rust, so if possible I want to replace z.lua with zoxide.

But before I start using zoxide, could you please tell me what are the differences between zoxide and z/z.lua, performance, feature, config, option, anything? Irrespective of rust language factor, what are the reasons for users of z/z.lua to use zoxide instead?

**Comment by ajeetdsouza** [comment, marked as answer]
Hey @c02y,

Regarding performance, zoxide (in my experiments) has generally been orders of magnitude faster than z.lua and other autojumpers. I try to keep zoxide as light as possible - heavy shell plugins tend to add up and slow down the shell startup and prompt.

Regarding features, z.lua does support a few features that are not in zoxide, simply because I haven't seen the need for it yet. For example, in z.lua, you can use `z -n` (where `n` is an integer) to go to the n<sup>th</sup> most "frecent" directory. I don't see the use for something that - I would personally just call `zi` and use the arrow keys to go to the desired directory. That said, if there's a feature you're missing from any other autojumper, do file an issue.

Regarding configurability, I think you'll find that both zoxide and z.lua are quite configurable. zoxide's CLI tends to be quite user-friendly, though - the `--help` menu usually tells you what you want to know. In fact, quite a few people actually write their own cd functions with custom behaviour, relying on zoxide's CLI only for the querying engine.

As a maintainer, I'm obviously biased towards zoxide, but I hope I was able to answer your question satisfactorily!

**Reply by c02y** [reply]
Got it. I've been using zoxide for a few days and I am quite happy with it, and I think I'll stick with it.

And thank you for this tool.

**Comment by mtalexan** [comment]
I know this has been answered, but I just stumbled on it myself.

The biggest difference is that the alternatives consider your current folder as context, while `zoxide` does not (nor does it have a way to).  The `z`/`z.lua` tools assume you want to take an action relative to your current folder (up, down, over), and use a global frecency list to help decide.  `Zoxide` is instead exclusively a global frecency jump list.

If you frequently need to jump between very different folders that don't share much of a common root path, `zoxide` is key.  No one else seems to do a good job of handling this without lots of extra typing (at which point tab completion on absolute paths may be faster to type than the other tools are to use).  
If however you usually want to jump into sub folders of the current one, or go back up to specific parent folder, `zoxide` probably isn't the right solution since it pays no attention to the directory you're currently in when doing searches.  

There's a very strong argument for using both `z`/`z.lua` and `zoxide` though.  For relative jumps you could use `z`/`z.lua` while non-relative jumps you can use `zoxide`.  Both have their places, and both are good at different things.

**Reply by ajeetdsouza** [reply]
I might consider adding support for relative jumps if there are enough requests for it. Personally, I don't find it useful - I prefer to describe where I want to be, and have the tool figure out how to get there.

**Reply by awerebea** [reply]
I agree, it would be nice to have some kind of `-l` flag to filter only child elements related to the current path.

**Reply by seeruk** [reply]
Maybe it could work without a flag, like `$ z . child dir`? (If it's necessary at all...)

**Reply by ajeetdsouza** [reply]
@seeruk that's exactly what I was planning for this feature - `z . foo` makes intuitive sense, as it's a path that starts from the local directory, and contains `foo`.
